File: src/main/java/io/gravitee/repository/management/model/Group.java
/**
 * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.gravitee.repository.management.model;

import java.util.Date;
import java.util.List;

/**
 * @author Nicolas GERAUD (nicolas.geraud at graviteesource.com)
 * @author GraviteeSource Team
 */
public class Group {

    /**
     * The group ID.
     */
    private String id;

    /**
     * the group name
     */
    private String name;

    /**
     * administrators of the group. May be null
     */
    private List<String> administrators;

    /**
     * events
     */
    private List<GroupEventRule> eventRules;

    /**
     * Creation date
     */
    private Date createdAt;

    /**
     * Last updated date
     */
    private Date updatedAt;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<String> getAdministrators() {
        return administrators;
    }

    public void setAdministrators(List<String> administrators) {
        this.administrators = administrators;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Date updatedAt) {
        this.updatedAt = updatedAt;
    }

    public List<GroupEventRule> getEventRules() {
        return eventRules;
    }

    public void setEventRules(List<GroupEventRule> eventRules) {
        this.eventRules = eventRules;
    }
}

File: gravitee-apim-console-webui/src/entities/management-api-v2/group/group.ts
/*
 * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export interface BaseGroup {
  id?: string;
  name?: string;
}

export type GroupEvent = 'API_CREATE' | 'APPLICATION_CREATE' | undefined;

export interface Group extends BaseGroup {
  eventRules?: GroupEvent[];
  manageable?: boolean;
  createdAt?: Date;
  updatedAt?: Date;
  maxInvitation?: boolean;
  apiRole?: string;
  lockApiRole?: boolean;
  applicationRole?: string;
  lockApplicationRole?: boolean;
  systemInvitation?: boolean;
  emailInvitation?: boolean;
  disableMembershipNotifications?: boolean;
  primaryOwner?: boolean;
}
